std-bits/base/src/main/java/org/enso/base/ObjectComparator.java

```diff
@@ -129,6 +131,10 @@ public int compare(Object thisValue, Object thatValue) throws ClassCastException
     }
 
     // Fallback to Enso
-    return fallbackComparator.apply(thisValue, thatValue);
+    try {
+      return fallbackComparator.apply(thisValue, thatValue);
+    } catch (PolyglotException ensoError) {
+      throw new ClassCastException(ensoError.getMessage());
+    }
```

Comment:
That's better but I'm still thinking if not too vague. If the `PolyglotException` inside is something like `IOError` or something, we probably do not want to swallow it the way we will swallow the CCE here.